server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/VariableNode.java

```diff
@@ -39,16 +40,14 @@
 import static org.eclipse.lsp.cobol.common.model.NodeType.VARIABLE_DEFINITION_NAME;
 
 /** The abstract class for all variable definitions. */
-@Getter
 @ToString(callSuper = true)
 @EqualsAndHashCode(callSuper = true)
 public abstract class VariableNode extends Node implements DefinedAndUsedStructure {
-
   public static final String PREFIX = "  ";
-  private final VariableType variableType;
-  private final String name;
-  @Setter private boolean global;
-  @EqualsAndHashCode.Exclude private final List<Location> usages = new ArrayList<>();
+  @Getter private final VariableType variableType;
+  @Getter private final String name;
+  @Getter @Setter private boolean global;
+  @EqualsAndHashCode.Exclude private final LinkedHashSet<Location> usages = new LinkedHashSet<>();
```

Comment:
Did you identify a situation where we ended up collecting duplicate Locations?

Reply:
If I remember correctly, it was caused by variable tree traverse on registration in symbol table, which was unnecessary.